runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/Engine.java

```diff
@@ -349,15 +351,18 @@ private final class ContextImpl implements EngineExtContext
         private final Configuration config;
         private final ErrorHandler errorHandler;
         private final Collection<DispatchAgent> dispatchers;
+        private final LabelManager labels;
 
         private ContextImpl(
             Configuration config,
             ErrorHandler errorHandler,
-            Collection<DispatchAgent> dispatchers)
+            Collection<DispatchAgent> dispatchers,
+            LabelManager labels)
```

Comment:
Let's pass in `supplyLabelId` as `ToIntFunction<String>` instead of the entire `LabelManager`.